core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -1252,6 +1253,26 @@ private static String toSql(RelNode root, SqlDialect dialect,
     relFn(relFn).optimize(rules, null).ok(expected);
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-4632">[CALCITE-4632]
+   * Find the least restrictive datatype for SARG</a>. */
+  @Test void testLeastRestrictiveTypeForSarg() {
```

Comment:
As discussed in the JIRA, the root cause is not from `RelToSqlConverter`, so I'm not sure this is the right place to add tests. How about we add more tests in `RexBuilderTest` and assert the `RexNode`'s type returned from `RexBuilder#makeIn`?

Reply:
ACK